src/overlays/actors/ovl_Bg_Hidan_Dalm/z_bg_hidan_dalm.c

```diff
@@ -21,17 +24,176 @@ const ActorInit Bg_Hidan_Dalm_InitVars = {
+    BgHidanDalm* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Dalm/BgHidanDalm_Destroy.s")
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    Collider_DestroyTris(globalCtx, &this->unk_16C);
+}
+
+void BgHidanDalm_WaitInteraction(BgHidanDalm* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((this->unk_16C.base.acFlags & 2) && !func_8008E988(globalCtx) &&
+        (player->swordAnimation == 22 || player->swordAnimation == 23)) {
+        this->unk_16C.base.acFlags &= ~2;
+        if (this->unk_16C.list[0].body.bumperFlags & 2 || this->unk_16C.list[1].body.bumperFlags & 2) {
+            this->dyna.actor.posRot.rot.y -= 0x4000;
+        } else {
+            this->dyna.actor.posRot.rot.y += 0x4000;
+        }
+        this->dyna.actor.posRot.pos.x += 32.5f * Math_Sins(this->dyna.actor.posRot.rot.y);
+        this->dyna.actor.posRot.pos.z += 32.5f * Math_Coss(this->dyna.actor.posRot.rot.y);
+
+        func_8002DF54(globalCtx, &this->dyna.actor, 8);
+        this->dyna.actor.flags |= 0x10;
+        this->actionFunc = BgHidanDalm_Shrink;
+        this->dyna.actor.bgCheckFlags &= ~2;
+        this->dyna.actor.bgCheckFlags &= ~8;
+        this->dyna.actor.speedXZ = 10.0f;
+        Flags_SetSwitch(globalCtx, this->switchFlag);
+        func_8002F7DC(&PLAYER->actor, NA_SE_IT_HAMMER_HIT);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_DARUMA_VANISH);
+    } else {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->unk_16C.base);
+    }
+}
+
+void BgHidanDalm_Shrink(BgHidanDalm* this, GlobalContext* globalCtx) {
+    static Vec3f accel = { 0, 0, 0 };
+    s32 i;
+    Vec3f velocity;
+    Vec3f pos;
+
+    if (Math_ApproxF(&this->dyna.actor.scale.x, 0.0f, 0.004f)) {
+        func_8002DF54(globalCtx, &this->dyna.actor, 7);
+        Actor_Kill(&this->dyna.actor);
+    }
+
+    this->dyna.actor.scale.y = this->dyna.actor.scale.z = this->dyna.actor.scale.x;
+
+    pos.x = this->dyna.actor.posRot.pos.x;
+    pos.y = this->dyna.actor.posRot.pos.y + this->dyna.actor.scale.x * 160.0f;
+    pos.z = this->dyna.actor.posRot.pos.z;
+
+    for (i = 0; i < 4; i++) {
+        velocity.x = 5.0f * Math_Sins(this->dyna.actor.posRot.rot.y + 0x8000) + (Math_Rand_ZeroOne() - 0.5f) * 5.0f;
+        velocity.z = 5.0f * Math_Coss(this->dyna.actor.posRot.rot.y + 0x8000) + (Math_Rand_ZeroOne() - 0.5f) * 5.0f;
+        velocity.y = (Math_Rand_ZeroOne() - 0.5f) * 1.5f;
+        func_80028B18(globalCtx, &pos, &velocity, &accel);
+    }
+}
+
+void BgHidanDalm_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanDalm* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    Actor_MoveForward(&this->dyna.actor);
+    func_8002E4B4(globalCtx, &this->dyna.actor, 10.0f, 15.0f, 32.0f, 5);
+}
+
+/*
+Update vertices of collider tris based on the current matrix
+*/
+void BgHidanDalm_UpdateCollider(BgHidanDalm* this) {
```

Comment:
```suggestion
/**
 * Update vertices of collider tris based on the current matrix
 */
void BgHidanDalm_UpdateCollider(BgHidanDalm* this) {
```

Reply:
is that the format for documentation comments?

Reply:
Yes